File: src/main/java/com/learnershi/rclasssocket/entity/ClassRoomStudyDataMap.java
package com.learnershi.rclasssocket.entity;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.HashMap;
import java.util.Map;

/**
 * classRoom 별 studyData
 */
@Accessors(chain = true)
@NoArgsConstructor
@Getter
@Setter
@Document
public class ClassRoomStudyDataMap {
    @Id
    private String classRoomId;     // ClassRoom Id

    private Map<Integer, StudyData> studyDataMap;

    public ClassRoomStudyDataMap setStudyData(StudyData studyData){
        if(this.studyDataMap == null){
            this.studyDataMap = new HashMap<>();
        }
        this.studyDataMap.put(studyData.getTabIndex(), studyData);
        return this;
    }

    @Builder
    public ClassRoomStudyDataMap(String classRoomId){
        this.classRoomId = classRoomId;
    }
}

File: src/main/java/com/learnershi/rclasssocket/entity/ClassRoomStudyDataMap.kt
package com.learnershi.rclasssocket.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

/**
 * classRoom 별 studyData
 */
@Document
class ClassRoomStudyDataMap(// ClassRoom Id
    @field:Id private val classRoomId: String
) {
    private var studyDataMap: MutableMap<Int, StudyData>? = null
    fun setStudyData(studyData: StudyData): ClassRoomStudyDataMap {
        if (studyDataMap == null) {
            studyDataMap = HashMap()
        }
        studyDataMap!![studyData.tabIndex] = studyData
        return this
    }
}